src/cluster.c

```diff
@@ -4522,12 +4523,14 @@ NULL
                         "configEpoch updated after importing slot %d", slot);
                 }
                 server.cluster->importing_slots_from[slot] = NULL;
+            }
+            clusterDelSlot(slot);
+            clusterAddSlot(n,slot);
+            if (n == myself && old != server.cluster->slots[slot]) {
```

Comment:
I'd prefer to move the add and delete slots up above the the previous conditional instead of duplicating the conditional logic. Similar to the original suggestion. 

```
            clusterDelSlot(slot);
            clusterAddSlot(n,slot);
            /* If this node was importing this slot, assigning the slot to
             * itself also clears the importing status. */
            if (n == myself &&
                server.cluster->importing_slots_from[slot])
            {
                /* This slot was manually migrated, set this node configEpoch
                 * to a new epoch so that the new version can be propagated
                 * by the cluster.
                 *
                 * Note that if this ever results in a collision with another
                 * node getting the same configEpoch, for example because a
                 * failover happens at the same time we close the slot, the
                 * configEpoch collision resolution will fix it assigning
                 * a different epoch to each node. */
                if (clusterBumpConfigEpochWithoutConsensus() == C_OK) {
                    serverLog(LL_WARNING,
                        "configEpoch updated after importing slot %d", slot);
                }
                server.cluster->importing_slots_from[slot] = NULL;

                /* After importing this slot, let the other nodes know as
                 * soon as possible. */
                clusterBroadcastPong(CLUSTER_BROADCAST_ALL);
            }

```

Reply:
It looks more concise that move the add and delete slots up.

Reply:
It looks more concise that move the add and delete slots up.